packages/paste-core/components/datagrid/src/DataGrid.tsx

```diff
@@ -0,0 +1,148 @@
+import * as React from 'react';
+import {Box} from '@twilio-paste/box';
+import type {BoxElementProps} from '@twilio-paste/box';
+import {useUID} from '@twilio-paste/uid-library';
+import {useCompositeState, Composite} from '@twilio-paste/reakit-library';
+import {Table} from '@twilio-paste/table';
+import type {TableProps} from '@twilio-paste/table';
+import {DataGridContext} from './DataGridContext';
+
+import {
+  ensureFocus,
+  getActiveElement,
+  getFirstFocusableIn,
+  getClosestCellFrom,
+  getClosestGridCellFromCurrentFocus,
+  isCell,
+  delayedSetFocusable,
+} from './utils';
+
+interface DataGridProps extends TableProps {
+  'aria-label': string;
+  element?: BoxElementProps['element'];
+}
+
+export const DataGrid = React.forwardRef<HTMLTableElement, DataGridProps>(({element = 'DATAGRID', ...props}, ref) => {
```

Comment:
It's not been a rule and I don't really know how to make it a rule, but where possible I've been trying to add more TSDoc comments on everything, including component exports so that you get a little more info in your VSCode when using them. 

I think it would be a nice pro touch to expand this across the system.